src/lib.rs

```diff
@@ -183,7 +182,8 @@ impl Decimal {
 
         let old_scale = self.scale();
 
-        if dp < old_scale && dp < 20 { // Technically, it's 28...
+        if dp < old_scale && dp < 20 {
+            // Technically, it's 28...
```

Comment:
This comment is not great, but loses context here. In fact, I think it's probably a bug which needs another issue created (e.g. why can't I round more than 20 dp?).